springboard/views.py

```diff
@@ -94,4 +106,35 @@ def api_notify(self):
     @notfound_view_config(renderer='springboard:templates/404.jinja2')
     def notfound(self):
         self.request.response.status = 404
-        return {}
+        return self.context()
+
+    @view_config(
+        route_name='locale_change',
+        renderer='springboard:templates/locale_change.jinja2')
+    def locale_change(self):
+        return self.context()
+
+    @view_config(route_name='locale')
+    @view_config(route_name='locale_matched')
+    def set_locale_cookie(self):
+        response = Response()
+        language = self.request.matchdict.get('language') or \
+            self.request.GET.get('language')
+
+        if language:
+            response.set_cookie('_LOCALE_', value=language, max_age=31536000)
```

Comment:
I have a slight preference for defining these numbers as constants somewhere like `ONE_YEAR = 31536000` and then change this to:
```python
response.set_cookie('_LOCALE_', value=language, max_age=ONE_YEAR)
```